### PR TITLE
T14904 - Fix Model events order and handle beforeDelete with SoftDelete

### PR DESCRIPTION
**Pull request by jenovateurs**
Hello!

* Type: bug fix | code quality
* Link to issue: phalcon/cphalcon#16980

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Small description of change:
I've changed the order of notification events. In that way, I fixed the problem between beforeDelete event and SoftDelete.
I also wrote tests for SoftDelete and Timestampable.

Thanks for the review!

### CODE REVIEW

**Comment by zsilbi** [comment]
@jenovateurs, could you please exclude the  `ext` directory?

**Comment by jenovateurs** [comment]
@zsilbi Have you a tip to always exclude the ext directory? 
And do you know the command to exclude the ext directory of this PR?
Maybe a git checkout ext upstream/v4.1.x?

**Comment by Jeckerson** [comment]
Also please rebase with `4.1.x` branch.

**Comment by jenovateurs** [comment]
Now I think it's a clean branch ;-).

**Review by zsilbi — APPROVED** [review]

File: tests/_data/fixtures/models/Invoices.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-declare(strict_types=1);
+declare(strict_types = 1);
```
Comment (Jeckerson) [inline tests/_data/fixtures/models/Invoices.php:12]:
```suggestion
declare(strict_types=1);
```

**Review by Jeckerson — APPROVED** [review]

**Pull request merged**